registration/views.py

```diff
@@ -403,9 +419,11 @@ def render_done(self, form, **kwargs):
         self.storage.reset()
         # also clean out the session
         clean_session(self.request.session)
-        clean_cobrand_session(self.request.session)
-        clean_widget_session(self.request.session)
-        clean_email_source_session(self.request.session)
+        # I Don't think we need to clear those session values on registration done, as it is going to be cleared on the
+        # next visit for Start page anyways.
+        # clean_cobrand_session(self.request.session)
+        # clean_widget_session(self.request.session)
```

Comment:
@danaspiegel We are cleaning the session whenever the start page loads. So, if someone came in via widget and then clicked on a shared link, he will go again to the start page, and there the session will be cleaned and setup with the new data based on the new URL.

So knowing that widgets and cobrand URLs will always lead to the Start page, then the session will always be cleaned and setup for each new visit.

I thought about all the cases, and tested it, and sure that this will not break anything, but yes, you are right, we still might miss some cases, so please let me know if a case came to your mind.